File: backend/app/schemas/project.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from app.models.project import ProjectStatus, ProjectMemberRole


class ProjectBase(BaseModel):
    name: str
    description: Optional[str] = None
    environment: Optional[str] = "production"
    tags: Optional[str] = None
    color: Optional[str] = "#30D158"


class ProjectCreate(ProjectBase):
    slug: str
    lob_id: str
    team_id: str
    component_id: Optional[str] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    status: Optional[ProjectStatus] = None
    environment: Optional[str] = None
    tags: Optional[str] = None
    color: Optional[str] = None
    component_id: Optional[str] = None


class ProjectResponse(ProjectBase):
    id: str
    slug: str
    lob_id: str
    team_id: Optional[str] = None
    team_name: Optional[str] = None
    component_id: Optional[str] = None
    component_name: Optional[str] = None
    status: ProjectStatus
    created_by: Optional[str]
    created_at: datetime
    updated_at: datetime
    connector_count: Optional[int] = 0
    healthy_count: Optional[int] = 0
    degraded_count: Optional[int] = 0
    down_count: Optional[int] = 0
    member_count: Optional[int] = 0

    class Config:
        from_attributes = True


class ProjectMemberCreate(BaseModel):
    user_id: str
    role: ProjectMemberRole = ProjectMemberRole.PROJECT_USER


class ProjectMemberUpdate(BaseModel):
    role: ProjectMemberRole


class ProjectMemberResponse(BaseModel):
    id: str
    project_id: str
    user_id: str
    role: ProjectMemberRole
    assigned_by: Optional[str]
    joined_at: datetime
    user_email: Optional[str] = None
    user_full_name: Optional[str] = None
    user_avatar_url: Optional[str] = None

    class Config:
        from_attributes = True

File: src/types/project.ts
export type ProjectMemberRole = 'project_admin' | 'project_user';

export interface ProjectMember {
  id: string;
  project_id: string;
  user_id: string;
  role: ProjectMemberRole;
  assigned_by?: string;
  joined_at: string;
  user_email?: string;
  user_full_name?: string;
  user_avatar_url?: string;
}

export interface Project {
  id: string;
  name: string;
  slug: string;
  description?: string;
  lob_id: string;
  team_id?: string;
  team_name?: string;
  component_id?: string;
  component_name?: string;
  status: 'active' | 'inactive' | 'maintenance' | 'archived';
  environment: string;
  tags?: string;
  color: string;
  created_by?: string;
  created_at: string;
  updated_at: string;
  connector_count: number;
  healthy_count: number;
  degraded_count: number;
  down_count: number;
  member_count: number;
}

export interface Component {
  id: string;
  name: string;
  slug: string;
  description?: string;
  color: string;
  icon: string;
  team_id: string;
  team_name?: string;
  lob_id: string;
  is_active: boolean;
  tenant_id: string;
  created_by?: string;
  created_at: string;
  updated_at: string;
  project_count?: number;
}
